tests/unit/models/keras/test_builders.py

```diff
@@ -12,53 +12,44 @@
 
 import pytest
 import tensorflow as tf
 import tensorflow_probability as tfp
 
 from tests.util.misc import empty_dataset
-from trieste.models.keras import GaussianNetwork, get_tensor_spec_from_data
+from trieste.models.keras import build_vanilla_keras_ensemble
 
 
-@pytest.mark.parametrize(
-    "query_point_shape, observation_shape",
-    [
-        ([1], [1]),
-        ([5], [1]),
-        ([5], [2]),
-    ],
-)
+@pytest.mark.parametrize("units, activation", [(10, "relu"), (50, tf.keras.activations.tanh)])
+@pytest.mark.parametrize("ensemble_size", [2, 5])
+@pytest.mark.parametrize("independent_normal", [False, True])
 @pytest.mark.parametrize("num_hidden_layers", [0, 1, 3])
-def test_gaussian_network_is_correctly_constructed(
-    query_point_shape: List[int], observation_shape: List[int], num_hidden_layers: int
+def test_build_vanilla_keras_ensemble(
+    ensemble_size: int,
+    num_hidden_layers: int,
+    units: int,
+    activation: Union[str, tf.keras.layers.Activation],
+    independent_normal: bool,
 ) -> None:
-    n_obs = 10
-    example_data = empty_dataset(query_point_shape, observation_shape)
-    query_points = tf.random.uniform([n_obs] + query_point_shape)
-
-    input_tensor_spec, output_tensor_spec = get_tensor_spec_from_data(example_data)
-    hidden_layer_args = []
-    for i in range(num_hidden_layers):
-        hidden_layer_args.append({"units": 10, "activation": "relu"})
-    network = GaussianNetwork(
-        input_tensor_spec,
-        output_tensor_spec,
-        hidden_layer_args,
+    example_data = empty_dataset([1], [1])
+    keras_ensemble = build_vanilla_keras_ensemble(
+        example_data,
+        ensemble_size,
+        num_hidden_layers,
+        units,
+        activation,
+        independent_normal,
     )
-    network_input, network_output = network.connect_layers()
-    network_built = tf.keras.Model(inputs=network_input, outputs=network_output)
-
-    # check input shape
-    assert network_input.shape[1:] == tf.TensorShape(query_point_shape)
-
-    # testing output shape is more complex as probabilistic layers don't have some properties
-    # we make some predictions instead and then check the output is correct
-    predictions = network_built.predict(query_points)
-    assert predictions.shape == tf.TensorShape([n_obs] + observation_shape)
 
-    # check layers
-    assert isinstance(network_built.layers[0], tf.keras.layers.InputLayer)
-    assert len(network_built.layers[1:-2]) == num_hidden_layers
-    assert isinstance(network_built.layers[-1], tfp.layers.DistributionLambda)
+    assert keras_ensemble.ensemble_size == ensemble_size
+    assert len(keras_ensemble.model.layers) == num_hidden_layers * ensemble_size + 3 * ensemble_size
+    if independent_normal:
+        assert isinstance(keras_ensemble.model.layers[-1], tfp.layers.IndependentNormal)
+    else:
+        assert isinstance(keras_ensemble.model.layers[-1], tfp.layers.MultivariateNormalTriL)
+    if num_hidden_layers > 0:
```

Comment:
his check shouldn't be necessary, right, given your previous assert about the number of layers?
```suggestion
```

Reply:
its needed after all as one of the inputs for this argument is 0, see line 28:
```
@pytest.mark.parametrize("num_hidden_layers", [0, 1, 3])
```